data/pigui/modules/saveloadgame.lua

```diff
@@ -98,50 +99,40 @@ local function showSaveFiles()
 end
 
 local function closeAndClearCache()
-	ui.closeCurrentPopup()
-	ui.showSavedGameWindow = nil
+	ui.saveLoadWindow:close()
+	ui.saveLoadWindow.mode = nil
 	saveFileCache = {}
 	popupOpened = false
 end
 
 local function closeAndLoadOrSave()
 	if selectedSave ~= nil and selectedSave ~= '' then
-		if ui.showSavedGameWindow == "LOAD" then
+		if ui.saveLoadWindow.mode == "LOAD" then
 			Game.LoadGame(selectedSave)
-		elseif ui.showSavedGameWindow == "SAVE" then
+		elseif ui.saveLoadWindow.mode == "SAVE" then
 			Game.SaveGame(selectedSave)
 		end
 		closeAndClearCache()
 	end
 end
 
-local function savedGameWindow()
-	if ui.showSavedGameWindow then
-		if not popupOpened then
-			popupOpened = true
-			ui.openPopup("LoadGame")
-		end
+ui.saveLoadWindow = ModalWindow.New("LoadGame", function()
+	local mode
+	mode = ui.saveLoadWindow.mode == 'SAVE' and lui.SAVE or lui.LOAD
+	optionTextButton(mode, nil, selectedSave~=nil, closeAndLoadOrSave)
```

Comment:
Minor nitpick, but please space around your operators like this. Also if selectedSave is guaranteed to be a "truthy" value when present, use `not selectedSave` instead; it's a little shorter and much easier for the reader to parse.
```suggestion
	optionTextButton(mode, nil, selectedSave ~= nil, closeAndLoadOrSave)
```

Reply:
I fixed the spacing, but now I'm wondering if the check shouldn't be changed to `selectedSave ~= nil and selectedSave ~= ''` like what is used in `closeAndLoadOrSave`

Reply:
I went ahead and added the `selectedSave ~= ''` check.